src/aggregate/aggregate_exec.c

```diff
@@ -443,12 +481,15 @@ static int execCommandCommon(RedisModuleCtx *ctx, RedisModuleString **argv, int
     if (rc != REDISMODULE_OK) {
       goto error;
     }
-  } else if (RunInThread(r) && 0 /* Temporary disabled */) {
-    // run in a thread
+  } else if (RunInThread(r)) {
+    RedisModuleBlockedClient *bc = RedisModule_BlockClient(ctx, NULL, NULL, NULL, 0);
+    // report block client start time
+    RS_CHECK_FUNC(RedisModule_BlockedClientMeasureTimeStart, bc);
+    r->sctx->redisCtx = RedisModule_GetThreadSafeContext(bc);
+    r->reqflags |= QEXEC_F_HAS_THCTX;
+    blockedClientReqCtx *BCRctx = blockedClientReqCtx_New(r, bc);
```

Comment:
I think it is better to move as much logic to the `AREQ_Execute_Callback` function as possible